### ISSUE TITLE
Use jll method to run binary

### ISSUE THREAD

**Issue by aviks** [MichaelHatherly/Tectonic.jl#1]
It's worth considering if the interface to this package should use the same method as the one used by jll packages. This would look like 
```
Tectonic.binary() do t
   run(`$t file.tex`)
end
```

The reason jll packages use this form is that it allows them the ability to manipulate the PATH and LD_LIBRARY_PATH before running the commands. Even though that might not be useful for this package now, I think its worth doing for two reasons. One, to ensure consistency across the ecosystem, and two, to keep open the possibility that the artifacts could be built and distributed by BinaryBuilder some day as jll packages.

**Comment by MichaelHatherly** [comment]
Yeah, that sounds good. I've not got up to speed on the jll packages yet so wasn't aware of that interface, seems pretty reasonable to change it now in anticipation. Any good examples/docs to look at regarding the interface, or is it as simple as

```julia
function Tectonic.binary(f)
    # ... setup
    f(path_to_binary)
    # ... clean up
end
```

> keep open the possibility that the artifacts could be built and distributed by BinaryBuilder some day as jll packages.

I gave `BinaryBuilder` a try first, but ran into some harfbuzz and icu issues that I couldn't solve with my limited understanding, but having this as a jll is the end goal once I, or someone else, manages to build from source.

**Comment by aviks** [comment]
> or is it as simple as

pretty much. That is effectively how the [jll wrappers](https://github.com/JuliaBinaryWrappers/pandoc_jll.jl/blob/master/src/wrappers/x86_64-linux-gnu.jl#L66) do it. 

> I gave BinaryBuilder a try first, but ran into some harfbuzz and icu issues

Yes, those kinds of things are not fun, an our rust stack is also not the most mature. I'm sure it'll get done eventually, however, as an interim, repackaging the prebuilt binaries is a reasonable strategy. [Here is an example](https://github.com/JuliaPackaging/Yggdrasil/blob/master/P/pandoc/build_tarballs.jl). The (admittedly minor) advantage is that you don't have to download/upload or maintain the artifacts.toml by hand.

**Comment by MichaelHatherly** [comment]
Thanks for the pointers.

**Comment by MichaelHatherly** [comment]
Fixed in 6bdfd0b080c9cd7d5667dd057fea94bd6a41e026.
